### ISSUE TITLE
video inference with sample rate

### ISSUE THREAD

**Issue by Kevin-Delnoije** [NVIDIA/DALI#1770]
I have a list of videos and for each video a subset of frames are begin fed as input to a neural network(pytorch).
having checked out the video/sequence examples and the superres training code,
it seems that a videoreader pipeline and maybe in combination with a Daligenericiterator should be able to do this.
But i am still wondering how to skip frames and get the current filename of the video being read.

in summary this is what i want to achieve:
-being able to skip frames
-keep track of current video name
-being able to iterate to next video
-convert frame to torch tensor
-model inference and repeat previous steps

video info:
300 to 600 frames per video
around 30k videos
 
any help is appreciated i am kinda stuck now.

**Comment by JanuszL** [comment]
Hi,
Please check [this example](https://docs.nvidia.com/deeplearning/sdk/dali-master-branch-user-guide/docs/examples/sequence_processing/video/video_file_list_outputs.html).
In the [`VideoReader` operator](https://docs.nvidia.com/deeplearning/sdk/dali-master-branch-user-guide/docs/examples/sequence_processing/video/video_file_list_outputs.html) you can provide a file list with the ranges of allowed frames (you can skip the one you don't like). If you assign a unique label to each file you can track file names.
Daligenericiterator should do in case of providing Torch tensors.
What is not possible is `being able to iterate to next video` as `VideoReader` (and mostly any other reader) expects a premade list of files, you cannot change it in the runtime. We know about this lack and plan to address it, but I cannot commit to any time frame yet.

**Comment by Kevin-Delnoije** [comment]
thanks,

If i understand correctly i need to prepare the file list in advance with corresponding end/start frame and label using the following format "/db/video/test_label.mp4 1 0 100\n".
The label format is a bit confusing to me, is it possible to provide a filename and a list with frame numbers to be read or something similar?
Also i am only interested in the filename instead of a label is there a property of the videoreader to access current filename of the video and detect when video ends so you know when frames of next video start .

**Comment by JanuszL** [comment]
Hi,
I understand your need but with the current DALI design what we have is most aligned with its design. Still, we would be more than happy to review any PR introducing such functionality.

**Comment by Kevin-Delnoije** [comment]
Hi,
Sure if you can give any pointers which files/functions to extend i will try to make a PR

**Comment by JanuszL** [comment]
@Kevin-Delnoije you can look into [that](https://github.com/NVIDIA/DALI/blob/master/dali/operators/reader/loader/video_loader.cc#L103) and [that](https://github.com/NVIDIA/DALI/blob/master/dali/operators/reader/loader/video_loader.h#L212) for the beggining.
